core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraFromRjsAdapter.kt

```diff
@@ -535,8 +536,60 @@ fun adaptRawInfra(infra: SignalingInfra, rjsInfra: RJSInfra): SimInfraAdapter {
         }
     }
 
-    // TODO: remove this stitching between new way of loading track-section info and previous way of
-    // loading network
+    // parse nodes
+    val switchTypeMap =
+        listOf(rjsInfra.switchTypes, RJSSwitchType.BUILTIN_NODE_TYPES_LIST)
+            .flatten()
+            .map { switchType -> Pair(switchType.id, switchType) }
+            .toMap()
+
+    val nodeNameToIdxMap = mutableMapOf<String, TrackNodeId>()
+    for (rjsNode in rjsInfra.switches) {
+        val ports = StaticPool<TrackNodePort, EndpointTrackSectionId>()
+        val portMap = mutableMapOf<String, TrackNodePortId>()
+        for ((rjsPortName, rjsPort) in rjsNode.ports) {
+            portMap[rjsPortName] =
+                ports.add(
+                    builder.getSectionEndpointIdx(
+                        rjsPort.track,
+                        Endpoint.fromEdgeEndpoint(rjsPort.endpoint)
+                    )
+                )
+        }
+        val switchType = switchTypeMap[rjsNode.switchType]
+        assert(switchType != null) {
+            "Assertion failed: Node ${rjsNode.id} references unknown switch-type ${rjsNode.switchType}"
+        }
```

Comment:
An assertion isn't quite right for this purpose, with default jvm config they're disabled and we would want an actual error here.

We can do something like that instead, it even makes the variable non-null from the start:
```suggestion
        val switchType = switchTypeMap[rjsNode.switchType]
            ?: throw RuntimeException("...")
```

(ideally we'd want a custom error type for *all* error cases, but these errors types are easier to refactor in bulk, later on)

Reply:
I actually chose to throw a (little bit) more specific [InfraHardLoadingError](https://github.com/osrd-project/osrd/pull/6799/commits/c4d0a6a9f7a45fa3697f8a392413181cd6cc6e5d#diff-0335c3a24d409feef2e7c3f5f3173cc4d08928c658e17faa8a0e71384922e6d3R561)